client/src/components/Auth/AuthForm.tsx

```diff
@@ -1,121 +1,184 @@
+  }
+
   const submitHandler = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
 
-    const email = emailInputRef.current?.value;
-    const password = passwordInputRef.current?.value;
-    const name = nameInputRef.current?.value;
-    console.log(email,name,password);
-    // optional: Add validation
+    // Add validation
+    if(!formIsValid){
+      return;
+    }
 
     setIsLoading(true);
     let url;
     if (isLogin) {
-      url =
-        'http://localhost:9000/user/login';
+      url = "http://localhost:9000/user/login";
     } else {
-      url =
-        'http://localhost:9000/user/register';
+      url = "http://localhost:9000/user/register";
     }
-    try 
-    {
-        const response = await Axios({
-            method: 'POST',
-            url: url,
-            data: {
-                name,
-                email,
-                password
-            }
-        });
-
-        if (response.status === 200)
-        {
-            // Could also use history.push('/login')
-            setIsLoading(false);
-            dispatch(authActions.login(response.data))
-            navigate('/');
-        }
-        else
-        {   
-            setIsLoading(false);
-            console.log("unspecified");
-        }
-    } 
-    catch (error) 
-    {
-        setIsLoading(false);
-        const errObj = error as AxiosError
-        const err = errObj.response?.data;
-        const response = err as AxiosError;
-        const notify = () => toast.error(response.message);
-        notify();
+    try {
+      const response = await Axios({
+        method: "POST",
+        url: url,
+        data: {
+          name,
+          email,
+          password,
+        },
+      });
 
+      if (response.status === 200) {
+        // Could also use history.push('/login')
+        setIsLoading(false);
+        dispatch(authActions.login(response.data));
+        showToastHandler('success','login Successful')
+        setTimeout(() => {
+          navigate("/");
+        }, 900);
+      } else {
+        setIsLoading(false);
+        console.log("unspecified");
+      }
+    } catch (error) {
+      setIsLoading(false);
+      const errObj = error as AxiosError;
+      const err = errObj.response?.data;
+      const response = err as AxiosError;
+      showToastHandler('fail',response?.message);
+      
     }
-}
+
+    nameResetHandler();
+    emailResetHandler();
+    passwordResetHandler();
+  };
+
+  const nameInputClasses = nameHasError ? `${classes["form-control"]} ${classes['invalid']}` : `${classes['form-control']}`;
+  const emailInputClasses = emailHasError ? `${classes["form-control"]} ${classes['invalid']}` : `${classes['form-control']}`;
+  const passwordInputClasses = passwordHasError ? `${classes["form-control"]} ${classes['invalid']}` : `${classes['form-control']}`;
 
   return (
+    <>
+    <Toast toastList={list} position='bottom-right' setList={setList} />
```

Comment:
Toast is now available app wide, also updated logic for showing toast. I've created a slice for toast and it can be triggered by dispatching showToast reducer function.